### PR TITLE
[Core] Improve CucumberOptions documentation

### PR DESCRIPTION
**Pull request by mpkorstanje**
## Summary

Improve CucumberOptions documentation

Fixes: #338

### CODE REVIEW

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 1]
```diff
@@ -6,65 +6,94 @@
...
+     * Treat undefined and pending steps as errors.
      */
     boolean strict() default false;
 
     /**
-     * @return the uris to the feature(s)
+     * Either a URI or path to a directory of features or a URI or path to a single
+     * feature optionally followed by a colon and line numbers.
+     * <p>
+     * When no feature path is provided Cucumber will use the package of the annotated
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: consider adding a comma for readability: "is provided, Cucumber"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 2]
```diff
@@ -6,65 +6,94 @@
...
+     * @see io.cucumber.core.model.FeatureWithLines
      */
     String[] features() default {};
 
     /**
-     * @return where to look for glue code (stepdefs and hooks)
+     * Package to load glue code (step definitions,
+     * hooks and plugins) from. E.g: {@code com.example.app}
+     * <p>
+     * When no glue is provided Cucumber will use the package of the annotated
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: consider adding a comma for readability: "is provided, Cucumber"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 3]
```diff
@@ -6,65 +6,94 @@
...
      */
     boolean strict() default false;
 
     /**
-     * @return the uris to the feature(s)
+     * Either a URI or path to a directory of features or a URI or path to a single
+     * feature optionally followed by a colon and line numbers.
+     * <p>
+     * When no feature path is provided Cucumber will use the package of the annotated
+     * class. So for example if the annotated class is {@code com.example.RunCucumber}
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: Consider changing "So for example if" to "For example, if"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 4]
```diff
@@ -6,65 +6,94 @@
...
      */
     String[] features() default {};
 
     /**
-     * @return where to look for glue code (stepdefs and hooks)
+     * Package to load glue code (step definitions,
+     * hooks and plugins) from. E.g: {@code com.example.app}
+     * <p>
+     * When no glue is provided Cucumber will use the package of the annotated
+     * class. So for example if the annotated class is {@code com.example.RunCucumber}
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: Consider changing "So for example if" to "For example, if"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 5]
```diff
@@ -6,65 +6,94 @@
...
+     * @see io.cucumber.core.model.GluePath
      */
     String[] glue() default {};
 
     /**
-     * @return where to look for glue code (stepdefs and hooks), in addition to default search path
+     * Package to load additional glue code (step definitions, hooks and
+     * plugins) from. E.g: {@code com.example.app}
+     * <p>
+     * These packages are used in addition to default described in {@code #glue}.
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: "to default" -> "to *the* default"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 6]
```diff
@@ -6,65 +6,94 @@
...
-     * @return whether or not to use monochrome output
+     * Don't colour terminal output.
      */
     boolean monochrome() default false;
 
     /**
-     * Specify a patternfilter for features or scenarios
-     *
-     * @return a list of patterns
+     * Only run scenarios whose names match regular expression.
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: Which regular expression? i.e. "the provided regular expression"

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 7]
```diff
@@ -6,65 +6,94 @@
...
      */
     String[] name() default {};
 
     /**
-     * @return what format should the snippets use. underscore, camelcase
+     * Naming convention for generated snippets.
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: Personally I'd prefer "format" over "naming" convention for this. And providing the options would be helpful.

File: core/src/main/java/cucumber/api/CucumberOptions.java [hunk 8]
```diff
@@ -6,65 +6,94 @@
...
      */
     SnippetType snippets() default SnippetType.UNDERSCORE;
 
     /**
-     * @return the options for the JUnit runner
+     * Pass the options to the JUnit module.
```
Comment (mlvandijk) [inline core/src/main/java/cucumber/api/CucumberOptions.java]: What is the difference to JUnit module and JUnit runner?

File: junit/src/main/java/cucumber/api/junit/Cucumber.java [hunk 9]
```diff
@@ -49,9 +49,10 @@
  * }
 Fail * </pre></blockquote>
  * <p>
- * Cucumber will look for a {@code .feature} file on the classpath, using the same resource
- * path as the annotated class ({@code .class} substituted by {@code .feature}).
- * <p>
+ * By default Cucumber will look for {@code .feature} and glue files on the classpath, using the same resource
+ * path as the annotated class. So for example if the annotated class is {@code com.example.RunCucumber} then
```
Comment (mlvandijk) [inline junit/src/main/java/cucumber/api/junit/Cucumber.java]: Consider changing "So for example if" to "For example, if"

File: junit/src/main/java/cucumber/api/junit/Cucumber.java [hunk 10]
```diff
@@ -49,9 +49,10 @@
...
+ * features and glue are assumed to be located in {@code com.example}.
+ * <p>d
```
Comment (mlvandijk) [inline junit/src/main/java/cucumber/api/junit/Cucumber.java]: Typo "d"

File: junit/src/main/java/cucumber/api/junit/Cucumber.java [hunk 11]
```diff
@@ -49,9 +49,10 @@
...
  * Additional hints can be given to Cucumber by annotating the class with {@link CucumberOptions}.
```
Comment (mlvandijk) [inline junit/src/main/java/cucumber/api/junit/Cucumber.java]: "given" -> "provided"

**Comment by mpkorstanje** [comment]
Cheers @mlvandijk!

**Pull request merged**